lib-impl/src/main/java/org/ergoplatform/appkit/impl/ErgoProverBuilderImpl.java

```diff
@@ -23,8 +21,7 @@
     private final BlockchainContextImpl _ctx;
 
     private ExtendedSecretKey _masterKey;
-    private DiffieHellmanTupleProverInput _firstSecret;
-    private DiffieHellmanTupleProverInput _secondSecret;
+    private DiffieHellmanTupleProverInput _dhtSecret;
```

Comment:
I assumed the proved can have both secrets, don't think we should limit the API to allow provers with single secret.

Reply:
I think (but hopefully others can chime in), that proveDHtuple makes sense for proving knowledge of only one of x or y, since the protocol will use only one of the secret even if the prover knows both. For proving knowledge of both, the prover must do two proveDlogs, one for each secret. Also, as far as I understand, the underlying sigma does not support two secrets for proveDHtuple.